osu.Game/Rulesets/Objects/Types/ComboIndex.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Rulesets.Objects.Types
+{
+    /// <summary>
+    /// Represents a manual and automated combo indices.
+    /// </summary>
+    public readonly struct ComboIndex : IEquatable<ComboIndex>
+    {
+        /// <summary>
+        /// The automated index value of the combo in relation to the beatmap.
+        /// </summary>
+        /// <remarks>
+        /// This is separated from <see cref="Manual"/> to disallow
+        /// manual combo colouring when not on the right skin to apply.
+        /// </remarks>
+        public readonly int Automated;
+
+        /// <summary>
+        /// The manual index value of the combo, this accounts for <see cref="IHasCombo.ComboOffset"/>.
+        /// </summary>
+        public readonly int Manual;
```

Comment:
...and this one to `WithOffset`.

Reply:
`Manual` goes well with `Ordinal` as this index can be set to literally anything, but since it should take value from `IHasCombo.ComboOffset`, `WithOffset` sounds fair.